tests/mnist_train_test.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import os
+import numpy.random as npr
+from examples import datasets
+from iree import runtime as iree_rt
+import jax.core
+import jax.numpy as jnp
+from jax import grad, random
+from jax.example_libraries import optimizers, stax
+from jax.example_libraries.stax import Dense, Relu, LogSoftmax
+from jax.tree_util import tree_flatten
+from iree.jax import (
+    like,
+    kernel,
+    IREE,
+    Program,
+)
+from tempfile import TemporaryDirectory
+import numpy as np
+from typing import Any, Callable
+import unittest
+
+
+def get_example_batch():
+  batch_size = 128
+  train_images, train_labels, test_images, test_labels = datasets.mnist()
+  num_train = train_images.shape[0]
+  num_complete_batches, leftover = divmod(num_train, batch_size)
+  num_batches = num_complete_batches + bool(leftover)
+
+  def data_stream():
+    rng = npr.RandomState(0)
+    while True:
+      perm = rng.permutation(num_train)
+      for i in range(num_batches):
+        batch_idx = perm[i * batch_size:(i + 1) * batch_size]
+        yield train_images[batch_idx], train_labels[batch_idx]
+
+  batches = data_stream()
+  return next(batches)
+
+
+def get_model():
```

Comment:
Thank you @sogartar for this pull request! It seems that this function duplicates with the one in https://github.com/iree-org/iree-jax/blob/main/examples/mnist_export.py. Is it a good idea to add a test for that file?

Reply:
Yes, I copied the model from there. I guess `mnist_export.py` example can benefit from a test too.